[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33173 [quant][graphmode] Add a pass to handle ops that doesn't require observation
* #33172 [quant][graphmode][refactor] Factor out insertDequantCall
* #33171 [quant][graphmode] refactor nodeQuantizable
* #32816 [quant][graphmode] Add add_relu pattern in skip values
* #32815 [quant][graphmode] Add asserts for Conv and ReLU modules
* #32814 [quant][graphmode] Skip quantizing input and output in matched module
* #32813 [quant][graphmode][refactor] Separate preprocess step for insertObserver
* #32812 [quant][graphmode][refactor] Change signature of getModuleAccessPath
* #32811 [quant][graphmode][refactor] test_jit.py - test_insert_observers_skip_values
* **#32810 [quant][graphmode][refactor] Change order of calling insertObservers for invoked methods**
* #32809 [quant][graphmode][refactor] Move the check for qconfig inside insertObserver call
* #32379 [quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32964 [jit] Add None parameter as parameter instead of attributes
* #32846 [jit][fix] Remove slot in parameter slot
* #32885 Add type annotation for bias in _ConvNd

Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods

Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit c7ee3f32:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=cccf5e7011f689c8dea499a1ab881900152e2575&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base cccf5e70 since Jan 29
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=c7ee3f329c5d3f375b8b5e54ccff09b086b18d66).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_linux_backward_compatibility_check_test](https://dr.pytorch.org/build-details.html?build_id=93011872) from Jan 29



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][refactor] Change order of calling insertObservers for invoked methods"


Summary:
insertObservers for invoked methods should be called after `addIntermediateValuesToSkipObserver`

Test Plan:
.

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
we are changing the way to insert observers, this is not useful anymore
